src/services/prisma-datastore/schema.prisma

```diff
@@ -0,0 +1,92 @@
+
+generator client {
+  provider = "prisma-client-js"
+  previewFeatures = ["extendedIndexes"]
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model Project {
+  id            Int       @id @default(autoincrement())
+  createdAt     DateTime  @default(now()) @map("created_at")
+  projectUID    String    @map("project_uid") @unique
+  name          String    @db.VarChar(50)
+  gateways      Gateway[]
+  @@map("project")
+}
+
+// A gateway comprises a sparrow device and a notecard. For now, the SparrowDevice itself isn't important
+model Gateway {
+  id            Int       @id @default(autoincrement())
+
+  // unique device uid in notehub
+  deviceUID    String     @unique @map("device_uid") @db.Char(64)
+
+  // device serial number (cached)
+  sn            String   @db.VarChar(50)
+
+  // the project the device belongs to
+  project       Project   @relation(fields:[project_id], references:[id])
+  project_id    Int
+
+  // the nodes paired with this gateway
+  nodes         Node[]
+  //radio         Node?
+
+  @@map("gateway")
+  @@index([project_id])   // surprised that relationships aren't automatically indexed  
+
+}
+
+model Node {
+  id              Int         @id @default(autoincrement())
+  
+  deviceEUI      String       @unique @map("device_eui") @db.Char(64)
+
+  name            String?     @db.VarChar(49) // cached
+  location        String?     @db.Char(15)    // cached
+  label           String?     @db.VarChar(50) // max 50
+  comment         String?     @db.VarChar(3000)  
+  lastSeenAt      DateTime?   @map("last_seen_at") // the last time an event was heard from this device
```

Comment:
```suggestion
  lastSeenAt      DateTime?   @map("last_seen_at") // the last time an event was heard from this device
```

I guess even if it's just a session, it'll leave a new event in _session.qo. Are we sure that's the right semantic meaning though? Maybe I'm splitting hairs.